storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4350,27 +4393,24 @@ void ha_rocksdb::load_auto_incr_value() {
   release_scan_iterator();
 }
 
-/* Get PK value from table->record[0]. */
-/*
-  TODO(alexyang): No existing support for auto_increment on non-pk columns, see
-  end of ha_rocksdb::create. Also see opened issue here:
-  https://github.com/facebook/mysql-5.6/issues/153
-*/
-void ha_rocksdb::update_auto_incr_val() {
+void ha_rocksdb::update_auto_incr_val(bool use_datadic, longlong target_value) {
   Field *field;
   longlong new_val;
   field = table->key_info[table->s->next_number_index].key_part[0].field;
+  if (use_datadic) {
+    new_val = target_value;
+  } else {
+    new_val = field->val_int();
+
+    // Merge the current in memory value into the auto_increment bookkeeping
+    // in the data dictionary
+    merge_auto_increment(new_val);
+  }
 
-  my_bitmap_map *const old_map =
-      dbug_tmp_use_all_columns(table, table->read_set);
```

Comment:
Is this code not needed for !use_datadic case above? It was originally here for handling index_last(). Is it not needed anymore?